syntax = "proto3";
package personalgram.auth;

service Auth {
	rpc login(LoginRequest) returns (AuthResponse);
	rpc logout(LogoutRequest) returns (AuthResponse);
	rpc currentUser(QueryUserRequest) returns (AuthResponse);
}

message LoginRequest {
	string username = 1;
	string password = 2;
}

message LogoutRequest {

}

message QueryUserRequest {
	
}

message AuthResponse {
	bool loggedIn = 1;
	string token = 2;
}
